Tribler/community/tunnel/hidden_community.py

```diff
@@ -186,9 +187,27 @@ def monitor_downloads(self, dslist):
             old_state = self.download_states.get(info_hash, None)
             state_changed = new_state != old_state
 
-            force_dht_lookup = (time.time() - self.last_dht_lookup.get(info_hash, 0)) >= 300
+            time_elapsed = (time.time() - self.last_dht_lookup.get(info_hash, 0))
+            force_dht_lookup = time_elapsed >= self.settings.dht_lookup_interval
+
+            # If the introducing circuit does not exist anymore or timed out: Build a new circuit
+            if info_hash in self.infohash_ip_circuits:
+                for circuit_id in self.infohash_ip_circuits[info_hash]:
+                    if circuit_id not in self.my_intro_points:
+                        self.infohash_ip_circuits[info_hash].remove(circuit_id)
+                        self._logger.debug('Recreate the introducing circuit for %s' % info_hash.encode('hex'))
+                        self.create_introduction_point(info_hash, required_exit=None)
```

Comment:
Nope. It will only get there once in the monitor_downloads, because after the disconnected circuit check the circuit is removed from `self.infohash_ip_circuits`. In my next commit it will be appended by the create_circuit() return value, so it is always containing the actual value when a new introduction circuit is created.
btw, same holds for rendezvous point circuits.